### PR TITLE
F/wpc 231 234 extensions and i18n support

### PR DESCRIPTION
**Pull request by DenisHuber**
Implementation of the following features (wip):
- API, Editor extensible
- Custom Content Models
- Multi-language support
- Reference Type implementation
- Browsers Credential Manager Feature implementation

### CODE REVIEW

File: packages/editor/src/client/routes/contentEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+            leftChildren={
+              <IconButtonLink
+                size={'lg'}
+                appearance="subtle"
+                icon={<Icon icon="angle-left" />}
+                route={ContentListRoute.create({type})}
+                onClick={e => {
+                  if (!unsavedChangesDialog()) e.preventDefault()
+                }}>
+                {t('articleEditor.overview.back')}
```
Comment (roosnic1) [inline packages/editor/src/client/routes/contentEditor.tsx:377]: i would avoid using translation strings from `articleEditor`. Also i think some of the translations in here are available in the globals.

File: packages/editor/src/client/routes/contentList.tsx [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+  const {t} = useTranslation()
+
+  useEffect(() => {
+    if (data?.content._all.list.nodes) {
+      setArticles(data?.content._all.list.nodes.map(a => a.content))
+    }
+  }, [data?.content._all.list])
+
+  function loadMore() {
```
Comment (roosnic1) [inline packages/editor/src/client/routes/contentList.tsx:67]: i strongly recommend using the the pagination components as in the article and page list. I already commented about this in the API. As soon as you want to sort the `fetchMore` is useless. And

File: packages/editor/src/client/routes/contentList.tsx [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+          <DescriptionList>
+            <DescriptionListItem label={t('articles.panels.title')}>
+              {currentContent?.title || t('articles.panels.untitled')}
+            </DescriptionListItem>
+
+            <DescriptionListItem label={t('articles.panels.createdAt')}>
+              {currentContent?.createdAt && new Date(currentContent.createdAt).toLocaleString()}
+            </DescriptionListItem>
+
+            {/* <DescriptionListItem label={t('articles.panels.updatedAt')}>
```
Comment (roosnic1) [inline packages/editor/src/client/routes/contentList.tsx:211]: why commented out?

File: packages/editor/src/client/routes/memberPlanList.tsx [hunk 4]
```diff
@@ -69,6 +56,84 @@ export function MemberPlanList() {
...
+            }}>
+            {' '}
+            {t('global.buttons.delete')}{' '}
+          </Button>
+        </Whisper>
+      </>
+    )
+  }
+
+  // const speaker = (
```
Comment (roosnic1) [inline packages/editor/src/client/routes/memberPlanList.tsx:118]: i think this can be deleted

File: packages/editor/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+export * from './client/client'
```
Comment (roosnic1) [inline packages/editor/src/index.ts:1]: why only those exports?

File: packages/editor/src/client/Editorcontext.tsx [hunk 6]
```diff
@@ -0,0 +1,4 @@
+import React from 'react'
+import {EditorConfig} from './interfaces/extensionConfig'
+
+export const ConfigContext = React.createContext<EditorConfig | undefined>(undefined)
```
Comment (roosnic1) [inline packages/editor/src/client/Editorcontext.tsx:4]: where is this used?

**Review by roosnic1 — CHANGES_REQUESTED** [review]

File: packages/api/src/business/contentModelBusiness.ts [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+      throw Error(`Schema ${identifier} not found`)
+    }
+    await validateInput({context: this.context}, schema.schema.content, input.content)
+    await validateInput({context: this.context}, schema.schema.meta, input.meta)
+
+    return this.context.dbAdapter.content.updateContent({
+      input: {
+        ...input,
+        contentType: identifier,
+        revision: 1,
```
[the remark this answers is not in the document]
Comment (DenisHuber) [reply packages/api/src/business/contentModelBusiness.ts:66]: revision handling is not yet implemented

File: packages/api/src/business/contentModelBusiness.ts [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+            if (Object.keys(schema.types).some(type => type === content?.contentType)) {
+              record = content
+            }
+          }
+        } catch (error) {}
+        if (!record) {
+          throw new Error(`Reference of type ${ref.contentType} and id ${ref.recordId} not valid`)
+        }
+
+        delete ref.record
```
[the remark this answers is not in the document]
Comment (DenisHuber) [reply packages/api/src/business/contentModelBusiness.ts:176]: It is checked if the reference is valid (check if record of this type and id exists in the db)

File: packages/api/src/business/contentModelBusiness.ts [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+export class BusinessLogic {
+  private context: Omit<Context, 'business'>
+
+  constructor(context: Omit<Context, 'business'>) {
+    this.context = context
+  }
+
+  async createContent(identifier: string, input: Content) {
+    const {roles} = this.context.authenticate()
+    authorise(CanCreateArticle, roles)
```
[the remark this answers is not in the document]
Comment (DenisHuber) [reply packages/api/src/business/contentModelBusiness.ts:27]: Will do

File: packages/api/src/db/content.ts [hunk 10]
```diff
@@ -0,0 +1,115 @@
+import {SortOrder, Limit, InputCursor, ConnectionResult} from './common'
+
+export enum DBContentState {
```
[the remark this answers is not in the document]
Comment (DenisHuber) [reply packages/api/src/db/content.ts:3]: Totally agree. Needs to be discussed for the upcoming feature implementations --> release flow & revisions

**Comment by roosnic1** [comment]
closing PR because not relevant anymore.

**Pull request closed without merging**